File: java/di/SettingsModule.java
package di;

import android.content.Context;

import data.settings.SettingsRepository;
import data.settings.SettingsRepositoryImpl;

// Hilt DI stub. Replace with @Module/@Provides when wiring Hilt.
public class SettingsModule {
    public static SettingsRepository provideSettingsRepository(Context context) {
        return new SettingsRepositoryImpl(context);
    }
}

File: kotlin/di/SettingsModule.kt
package di

import android.content.Context
import data.settings.SettingsRepository
import data.settings.SettingsRepositoryImpl
import data.settings.SettingsStore

// Hilt DI stub module placeholder. Replace with @Module and @Provides as needed.
object SettingsModule {
    fun provideSettingsRepository(context: Context): SettingsRepository =
        SettingsRepositoryImpl(SettingsStore(context))
}

